backend/src/packages/chats/chat.repository.ts

```diff
@@ -21,12 +21,21 @@ class ChatRepository implements Repository {
     return Promise.resolve([]);
   }
 
-  public async findAllByUserId(userId: number): Promise<ChatEntity[]> {
+  public async findAllByUserId(
+    userId: number,
+    query: string,
+  ): Promise<ChatEntity[]> {
     const chats = await this.chatModel
       .query()
       .withGraphJoined(ChatsRelation.MEMBERS)
       .whereExists(UserToChatModel.query().where('userId', userId))
       .orderBy('createdAt', 'DESC')
+      .modify((builder) => {
+        if (query) {
+          void builder.whereILike('name', `%${query}%`);
```

Comment:
Prevent sql injection

Reply:
@v0ldemar01 this query build to this sql request and name is using prepared statement, so should i leave it or try to find how to avoid %${query}%? And also in knex documentation write like in my code. Link to [docs](https://knexjs.org/guide/query-builder.html#wherelike)

![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92929504/c98af7c8-fdca-4282-bd89-551b32c2c315)

Reply:
https://stackoverflow.com/questions/50327070/how-do-i-escape-in-knex-where-like-query

Reply:
@v0ldemar01 is this right variant:
      .modify((builder) => {
        if (query) {
          void builder.where('title', 'iLike', `%${query.replaceAll('%', '\\%')}%`);
        }
      })
      ?
As I understand SQL injection can be in case inserting value is in the end of SQL query, isn`t it?

Reply:
<img width="521" alt="image" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/43109766/89abcef4-7ff9-4918-9bca-8cc28dd51d80">

Reply:
@v0ldemar01 it does not work

Reply:
@v0ldemar01 this case does not work

Reply:
https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92929504/76fa21ff-7add-4a15-b517-ad9c58193cbb

Reply:
<img width="829" alt="image" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/43109766/999ef1a3-cece-49a2-ac58-072e0104ae5a">

Reply:
@v0ldemar01 i found that this variant:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92929504/0ad80c31-4895-4c38-8f70-45be5ea5edfd)
is safety because sql reder its directly to '?', also it does the same for findById
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92929504/4b49563d-91c4-428c-9061-954f22771b9d)

Reply:
Okey

Reply:
done